From f4bdfe5babd6a1e2ee403e35b393453c0b9f11ec Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Mon, 3 Jul 2023 13:26:23 +0200
Subject: [PATCH 08/20] net: dsa: qca8k: on fdb dump treat default port VID as
 0

DSA have no concept of switch requiring a specific port VID when VID 0
is requested. The driver handle this internally by replacing VID 0 with
the default value. Currently fdb dump doesn't handle this case and
return the actually VID set in the fdb table.

Fix this special case by replacing back the default VID with 0 on
reading the table.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/net/dsa/qca/qca8k-common.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/net/dsa/qca/qca8k-common.c
+++ b/drivers/net/dsa/qca/qca8k-common.c
@@ -840,10 +840,14 @@ int qca8k_port_fdb_dump(struct dsa_switc
 
 	mutex_lock(&priv->reg_mutex);
 	while (cnt-- && !qca8k_fdb_next(priv, &_fdb, port)) {
+		u16 vid = _fdb.vid;
+
 		if (!_fdb.aging)
 			break;
 		is_static = (_fdb.aging == QCA8K_ATU_STATUS_STATIC);
-		ret = cb(_fdb.mac, _fdb.vid, is_static, data);
+		if (_fdb.vid == QCA8K_PORT_VID_DEF)
+			vid = 0;
+		ret = cb(_fdb.mac, vid, is_static, data);
 		if (ret)
 			break;
 	}
